src/main/go/args/java_execution.go

```diff
@@ -14,6 +15,24 @@ func getArchSpecificDirectory() string {
 	return fmt.Sprintf("%s-%s", runtime.GOOS, runtime.GOARCH)
 }
 
+func DetectJvmVersion(javaBin string) (string, error) {
+	command := exec.Command(javaBin, "--version")
+	if err := command.Run(); err != nil {
+		return "", err
+	}
+	output, err := command.Output()
+	if err != nil {
+		return "", fmt.Errorf("could not exec java to determine jvm version: %w", err)
+	}
+
+	regex := regexp.MustCompile("version \"(.*)\"")
+	matches := regex.FindStringSubmatch(string(output))
```

Comment:
This is not going to work for patch versions:

```
openjdk version "23.0.1" 2024-10-15
OpenJDK Runtime Environment Temurin-23.0.1+11 (build 23.0.1+11)
OpenJDK 64-Bit Server VM Temurin-23.0.1+11 (build 23.0.1+11, mixed mode, sharing)
```

I don't see anything in this method or caller that converts "23.0.1" to "23" before it looks it up in the map.

Reply:
You are right, my bad. Added